src/mpi/coll/src/csel.c

```diff
@@ -0,0 +1,1253 @@
+            else if (!strcmp(str, "ineighbor_alltoall"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__INEIGHBOR_ALLTOALL;
+            else if (!strcmp(str, "ineighbor_alltoallv"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__INEIGHBOR_ALLTOALLV;
+            else if (!strcmp(str, "ineighbor_alltoallw"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__INEIGHBOR_ALLTOALLW;
+            else if (!strcmp(str, "ireduce"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__IREDUCE;
+            else if (!strcmp(str, "ireduce_scatter"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__IREDUCE_SCATTER;
+            else if (!strcmp(str, "ireduce_scatter_block"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__IREDUCE_SCATTER_BLOCK;
+            else if (!strcmp(str, "iscan"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__ISCAN;
+            else if (!strcmp(str, "iscatter"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__ISCATTER;
+            else if (!strcmp(str, "iscatterv"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__ISCATTERV;
+            else if (!strcmp(str, "neighbor_allgather"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__NEIGHBOR_ALLGATHER;
+            else if (!strcmp(str, "neighbor_allgatherv"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__NEIGHBOR_ALLGATHERV;
+            else if (!strcmp(str, "neighbor_alltoall"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__NEIGHBOR_ALLTOALL;
+            else if (!strcmp(str, "neighbor_alltoallv"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__NEIGHBOR_ALLTOALLV;
+            else if (!strcmp(str, "neighbor_alltoallw"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__NEIGHBOR_ALLTOALLW;
+            else if (!strcmp(str, "reduce"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__REDUCE;
+            else if (!strcmp(str, "reduce_scatter"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__REDUCE_SCATTER;
+            else if (!strcmp(str, "reduce_scatter_block"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__REDUCE_SCATTER_BLOCK;
+            else if (!strcmp(str, "scan"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__SCAN;
+            else if (!strcmp(str, "scatter"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__SCATTER;
+            else if (!strcmp(str, "scatterv"))
+                tmp->u.collective.coll_type = MPIR_CSEL_COLL_TYPE__SCATTERV;
+            else
+                MPIR_Assert(0);
+        } else if (!strcmp(ckey, "comm_size=pow2")) {
+            tmp->type = CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_POW2;
```

Comment:
I guess here we can refactor with a routine, e.g. `op_type_str_to_id`.

Reply:
Unless there's an automatic way to do this (and I don't mean with a script), it's the same amount of code to be maintained.

Reply:
Refactor sometime is not just about the amount of code, it is also about making semantic contexts cleaner, which can result in easier maintenance. But I agree that it is not that important here.